kuzzleio/kuzzle
# Apply default mapping when putting collection in cache at startup

## What does this PR do ?

This PR improve the support of index/collection created directly through Elasticsearch.  
When you start Kuzzle, the existent index/collection will be put in the cache system and the default mapping will be applied on each one.  

`InternalEngine.updateMapping()` has now an optional parameter `index` to update other mappings than the internal one.  
I didn't use `ElasticSearch.updateMapping()` because the function doesn't allow to modify internal mapping.

Fix https://github.com/kuzzleio/kuzzle/issues/1109

I started to write functional test with Cucumber to reproduce the bug but I had to restart the Kuzzle instance If I want my Elasticsearch collection to be available in Kuzzle.. (see https://github.com/kuzzleio/kuzzle/issues/1111)

### How should this be manually tested?
  - Step 1 : Download PR scripts : `wget -O pr_files.zip https://gist.github.com/Aschen/da7c243bbc8acbba08995973d72fdff3/archive/c436b83891266b8b527c44403e687dc282f4e61b.zip && unzip pr_files.zip && mv da7c243*/* . && rm pr_files.zip`
  - Step 2 : Start docker-compose with Elastic listening on 9200 : `docker-compose -f docker-compose-pr.yml up`
  - Step 3 : Create the index with Elasticsearch: `node es-index.js` 
  - Step 4 :  Create a document with Elasticsearch: `node es-document.js`
  - Step 5 : Restart Kuzzle to recreate cache and apply default mapping : `docker-compose -f docker-compose-pr.yml restart kuzzle`
  - Step 6 : Update the document with Kuzzle : `node kz-update.js <document id> Jean`
  - Step 7 : Check the `_kuzzle_info` mapping : `node es-mapping.js`

### Boyscout
 
 - Atom removed all spaces in `lib/services/elasticsearch.js` (:
